collections/comparators.ts

```diff
@@ -0,0 +1,11 @@
+/** This module is browser compatible. */
+
+/** Compares its two arguments for ascending order using JavaScript's built in comparison operators. */
+export function ascend<T>(a: T, b: T) {
+  return a < b ? -1 : a > b ? 1 : 0;
+}
+
+/** Compares its two arguments for descending order using JavaScript's built in comparison operators. */
+export function descend<T>(a: T, b: T) {
+  return a < b ? 1 : a > b ? -1 : 0;
+}
```

Comment:
I think this module is not needed; if it is used in `binary_heap.ts` we could just define those inline in that module.

Reply:
When I add BSTree and RBTree in separate PRs, they will use these comparators too for determining how to sort the values added to the tree. That's why I put them in their own file. These functions work with `Array.prototype.sort` too. I'm fine with moving them too though. Having them at least somewhere is useful to avoid having to re-implement these functions whenever you want to change the default sort behavior of a data structure or sort an existing one.

Reply:
Here is a usage example for making the BinaryHeap be a Min Heap instead of a Max Heap.

```ts
import { BinaryHeap, ascend } from "https://deno.land/std@0.129.0/collections/mod.ts";
const maxHeap = new BinaryHeap<number>(); // descend is default compare function.
const minHeap = new BinaryHeap<number>(ascend);
```

Reply:
Another option would be to keep them in their own separate file but also update binary_heap.ts to re-export them for easy access, that way you could import them all from binary_heap.ts instead of importing the entire mod.ts or having 2 separate import lines.

Reply:
> Another option would be to keep them in their own separate file but also update binary_heap.ts to re-export them for easy access

This sounds better option to me. And also I think it's better to prefix this module as `_comparators.ts` to prevent the direct dependency to this file.

Reply:
I added an underscore prefix to the `comparators.ts` and `comparators_test.ts` as requested and updated `binary_heap.ts` to export them.

I've also updated the README.md file to have a section for BinaryHeap, showing the efficiency of the functions. I included an example of both min and max heaps along with showing how the comparator functions can be used to compare properties of the values in a BinaryHeap.